File: old_baser/javarecipe/src/main/java/work/shion/javarecipe/pages/entrypoint/presenters/MainPresenter.java
package work.shion.javarecipe.pages.entrypoint.presenters;

import android.os.Handler;
import android.os.HandlerThread;
import android.os.Looper;

import com.annimon.stream.Optional;

import java.lang.ref.WeakReference;

import work.shion.javarecipe.pages.entrypoint.contracts.MainPresenterContract;
import work.shion.javarecipe.pages.entrypoint.contracts.MainViewContract;
import work.shion.javarecipe.pages.entrypoint.models.MainModel;


/**
 * メインコンテンツの挙動実装
 */
public class MainPresenter implements MainPresenterContract {
    private final MainModel model;
    private final WeakReference<MainViewContract> viewer;


    public MainPresenter(
            MainModel model,
            WeakReference<MainViewContract> viewer
    ) {
        this.model = model;
        this.viewer = viewer;
    }


    /**
     * 最初に表示する画面の呼び出し
     */
    @Override
    public void callFirstView() {
        HandlerThread subThread = new HandlerThread("sub");
        subThread.start();
        new Handler(subThread.getLooper()).postDelayed(() -> {
            final int random = model.getRandom();
            Optional.ofNullable(viewer.get())
                    .ifPresent(viewer -> {
                        new Handler(Looper.getMainLooper()).post(() -> {
                            if (random < 50) {
                                viewer.goTutorialPage();
                            } else {
                                viewer.goTabPage();
                            }
                        });
                    });
        }, 1500);
    }
}

File: ktrecipe/src/main/java/work/shion/ktrecipe/pages/entrypoint/presenters/MainPresenter.kt
package work.shion.ktrecipe.pages.entrypoint.presenters

import android.os.Handler
import android.os.HandlerThread
import android.os.Looper
import work.shion.ktrecipe.pages.entrypoint.contracts.MainPresenterContract
import work.shion.ktrecipe.pages.entrypoint.contracts.MainViewContract
import work.shion.ktrecipe.pages.entrypoint.models.MainModel
import java.lang.ref.WeakReference


/**
 * メインコンテンツの挙動実装
 */
class MainPresenter(
        private val model: MainModel,
        private val viewer: WeakReference<MainViewContract>
) : MainPresenterContract {

    /**
     * 最初に表示する画面の呼び出し
     */
    override fun callFirstView() {
        val subThread = HandlerThread("sub").apply { start() }.looper
        Handler(subThread).postDelayed({
            val random = model.getRandom()
            viewer.get()?.apply {
                Handler(Looper.getMainLooper()).post {
                    if (random < 50) {
                        goTutorialPage()
                    } else {
                        goTabPage()
                    }
                }
            }
        }, 1500)
    }
}
